core-test/src/test/scala/org/bitcoins/core/crypto/SchnorrTest.scala

```diff
@@ -0,0 +1,166 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoin.NativeSecp256k1
+import org.bitcoins.core.util.{CryptoUtil, NumberUtil}
+import org.bitcoins.testkit.core.gen.{CryptoGenerators, NumberGenerator}
+import org.bitcoins.testkit.util.BitcoinSUnitTest
+import scodec.bits.ByteVector
+
+class SchnorrTest extends BitcoinSUnitTest {
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    generatorDrivenConfigNewCode
+
+  behavior of "NativeSecp256k1"
+
+  it must "act like a normal schnorrSign if the normal nonce is specified" in {
+    forAll(CryptoGenerators.nonZeroPrivKey, NumberGenerator.bytevector(32)) {
+      case (privKey, message) =>
+        val nonce = SchnorrNonce.fromBipSchnorr(privKey, message)
+
+        assert(nonce.bytes.toArray.exists(_ != 0.toByte))
+
+        val sigWithNonce =
+          NativeSecp256k1.schnorrSignWithNonce(message.toArray,
+                                               privKey.bytes.toArray,
+                                               nonce.bytes.toArray)
+
+        assert(
+          SchnorrDigitalSignature(ByteVector(sigWithNonce)) ==
+            Schnorr.signWithNonce(message, privKey, nonce)
+        )
+
+        val sig =
+          NativeSecp256k1.schnorrSign(message.toArray, privKey.bytes.toArray)
+
+        assert(
+          SchnorrDigitalSignature(ByteVector(sig)) ==
+            Schnorr.sign(message, privKey)
+        )
+
+        assert(ByteVector(sigWithNonce) == ByteVector(sig))
+    }
+  }
+
+  it must "correctly compute a Schnorr public key given an R value" in {
+    forAll(CryptoGenerators.nonZeroPrivKey, NumberGenerator.bytevector(32)) {
+      case (privKey, message) =>
+        val publicKey: ECPublicKey = privKey.publicKey
+        val messageArr: Array[Byte] = message.toArray
+
+        val nonce = SchnorrNonce.fromBipSchnorr(privKey, message)
+
+        assert(nonce.bytes.toArray.exists(_ != 0.toByte))
+
+        val rBytes = nonce.publicKey.bytes
+
+        val sig: Array[Byte] =
+          NativeSecp256k1.schnorrSign(messageArr, privKey.bytes.toArray)
+
+        val (rKey, s) = sig.splitAt(32)
+
+        assert(rBytes.tail == ByteVector(rKey))
+
+        val sPub: Array[Byte] =
+          NativeSecp256k1.computeSchnorrPubKey(messageArr,
+                                               rBytes.toArray,
+                                               publicKey.bytes.toArray)
+        val sPubKey: ECPublicKey = ECPublicKey.fromBytes(ByteVector(sPub))
+
+        assert(
+          sPubKey == Schnorr.computePubKey(message, nonce.publicKey, publicKey))
+
+        val realSPriv = ECPrivateKey.fromBytes(ByteVector(s))
+        val realSPub = realSPriv.publicKey
+
+        assert(sPubKey == realSPub)
+    }
+  }
+
+  it should "leak keys if two messages are signed" in {
+    // The order of the secp256k1 curve
```

Comment:
There needs to be more context added to this. Why should we be able to leak keys if two messages are signed? At least a link to why this is the case, or a paragraph explanation.